rfcs/0005-pablo-distribution.md

```diff
@@ -0,0 +1,667 @@
+← (2)
+
+Now with (1) + (2),
+
+<img src="0005-pablo-distribution-assets/images/stem-6a000622e842e98de57502915826da7b.png" width="245" height="20" alt="stem 6a000622e842e98de57502915826da7b" /></span>
+
+Therefore, same computation as before with <img src="0005-pablo-distribution-assets/images/stem-9849cee8ec3e29bf6d2ea80a64d995dd.png" width="23" height="10" alt="stem 9849cee8ec3e29bf6d2ea80a64d995dd" /></span>
+number of shares added to the staker position works as expected.
+
+##### 5.3.1.5. When splitting an existing position
+
+As the total reward pool is not affected the splitting is just creating
+a new position using some ratio. If the ratio is <img src="0005-pablo-distribution-assets/images/stem-603de94498e154610e3066ec63603017.png" width="25" height="9" alt="stem 603de94498e154610e3066ec63603017" /></span>
+From (1)
+
+First position <img src="0005-pablo-distribution-assets/images/stem-ed2c175456fa8dcae30f92f61b3694ff.png" width="151" height="20" alt="stem ed2c175456fa8dcae30f92f61b3694ff" /></span>
+
+Second position <img src="0005-pablo-distribution-assets/images/stem-706fbeec167ddb5aeb84ef0c7bde2f57.png" width="192" height="20" alt="stem 706fbeec167ddb5aeb84ef0c7bde2f57" /></span>
+
+Summing these positions would give the original position(equation 1) as
+the ratio terms cancel out.
+
+As this method uses a reward pooling based approach to calculate the
+rewards for each staker out of it on-demand, rest of the document refers
+to this as the "reward pooling(**RP**) based approach".
+
+#### 5.3.2. Data Structures
+
+Staking rewards pallet already uses the following fNFT data structure,
+
+    /// staking typed fNFT, usually can be mapped to raw fNFT storage type
+    #[derive(Debug, PartialEq, Eq, Copy, Clone, Encode, Decode, TypeInfo)]
+    pub struct StakingNFT<AccountId, AssetId, Balance, Epoch, Rewards> {
+        /// The staked asset.
+        pub asset: AssetId,
+        /// The original stake this NFT was minted for.
+        pub stake: Balance,
+        /// The reward epoch at which this NFT will start yielding rewards.
+        pub reward_epoch_start: Epoch,
+        /// List of reward asset/pending rewards.
+        pub pending_rewards: Rewards,
+        /// The date at which this NFT was minted.
+        pub lock_date: Timestamp,
+        /// The duration for which this NFT stake was locked.
+        pub lock_duration: DurationSeconds,
+        /// The penalty applied if a staker unstake before the end date.
+        pub early_unstake_penalty: Penalty<AccountId>,
+        /// The reward multiplier.
+        pub reward_multiplier: Perbill,
+    }
+
+This data structure can be modified to support the RP approach by
+adding/adjusting/removing some fields.
+
+    /// This can possibly be renamed as `RewardPoolPosition`
+    pub struct StakingNFT<RewardPoolId, AccountId, AssetId, Balance> {
```

Comment:
hm, other struct was called StakingNFT too....

Reply:
Yup, we can later modify based on any new naming, as long as the meaning is obvious this is ok